app/controllers/api/v1/messages_controller.rb

```diff
@@ -58,8 +62,38 @@ def mark_read
         render json: @message, serializer: serializer
       end
 
+      api :PUT, "/v1/messages/mark_all_read", "Mark all messages as read"
+      def mark_all_read
+        @subscriptions = Subscription.unread.of_active_user
+        if params[:scope].present?
+          @subscriptions = apply_scope(@subscriptions.joins(:message), params[:scope])
+        end
+        @subscriptions.update_all state: 'read'
+        render json: {}
+      end
+
       private
 
+      def apply_scope(records, scope)
+        return records unless ALLOWED_SCOPES.include? scope
+        records.where("messages.#{scope}_id IS NOT NULL")
+      end
+
+      def paginate_and_render(records)
+        meta = {}
+        if params[:page].present?
+          records = records
+            .page(params["page"])
+            .per(params["per_page"] || DEFAULT_SEARCH_COUNT)
```

Comment:
Good to put an upper limit on `per_page` and `page`... e.g. query param could be `?per_page=100000`

Reply:
that's not a bad idea, any idea how that could be implemented globally ? (guessing a concern)

Reply:
We already have, check  out https://github.com/crossroads/api.goodcity/blob/master/app/controllers/api/v1/api_controller.rb#L53

so you can just change the line to `.per(per_page)` and it will grab the param and check it for you

Reply:
@steveyken Cool, although that doesn't look right, `DEFAULT_SEARCH_COUNT` should not be the maximum